CHANGELOG.md

```diff
@@ -4,7 +4,215 @@ All notable changes to this project will be documented in this file.
+            // Handle any errors:
+            println!("Error submitting tx: {message}");
+        },
+        // Continue otherwise:
+        _ => continue,
+    }
+}
+```
+
+### Subxt-codegen: Tidy crate interface ([#1225](https://github.com/paritytech/subxt/pull/1225))
+
+The `subxt-codegen` crate has always been a bit of a mess because it wasn't really supposed to be used outside of the subxt crates, which had led to issues like https://github.com/paritytech/subxt/issues/1211.
+
+This PR tidies up the interface to that crate so that it's much easier now to programmatically generate the Subxt interface. Now, we have three properly supported ways to do this, depending on your needs:
+
+1. Using the `#[subxt]` macro.
+2. Using the `subxt codegen` CLI command.
+3. Programmatically via the `subxt-codegen` crate.
+
+Each method aims to expose a similar and consistent set of options.
+
+If you were previously looking to use parts of the type generation logic to, for instance, generate runtime types but not the rest of the Subxt interface, then the https://github.com/paritytech/scale-typegen crate will aim to fill this role eventually.
+
+That sums up the most significant changes. A summary of all of the relevant changes is as follows:
+
+### Added
+
+- CLI: Add command to fetch chainSpec and optimize its size ([#1278](https://github.com/paritytech/subxt/pull/1278))
+- Add legacy RPC usage example ([#1279](https://github.com/paritytech/subxt/pull/1279))
+- RPC: Implement legacy RPC system_account_next_index ([#1250](https://github.com/paritytech/subxt/pull/1250))
+- Lightclient: Add support for configuring multiple chains ([#1238](https://github.com/paritytech/subxt/pull/1238))
+- Extrinsics: Allow static decoding of signed extensions ([#1235](https://github.com/paritytech/subxt/pull/1235))
+- Extrinsics: Support decoding signed extensions ([#1209](https://github.com/paritytech/subxt/pull/1209))
+- ChargeAssetTxPayment: Add support for generic AssetId (eg `u32` or `MultiLocation`) ([#1227](https://github.com/paritytech/subxt/pull/1227))
+- Add Clone + Debug on Payloads/Addresses, and compare child storage results ([#1203](https://github.com/paritytech/subxt/pull/1203))
+
+### Changed
+
+- Lightclient: Update smoldot to `0.14.0` and smoldot-light to `0.12.0` ([#1307](https://github.com/paritytech/subxt/pull/1307))
+- Cargo: Switch to workspace lints ([#1299](https://github.com/paritytech/subxt/pull/1299))
+- Update substrate-* and signer-related dependencies ([#1297](https://github.com/paritytech/subxt/pull/1297))
+- Fix lightclient WASM tests ([#1287](https://github.com/paritytech/subxt/pull/1287))
+- Change SignedExtension matching logic and remove SkipCheckIfFeeless bits ([#1283](https://github.com/paritytech/subxt/pull/1283))
+- Update the README with the new location of node-cli ([#1282](https://github.com/paritytech/subxt/pull/1282))
+- impl RpcClientT for `Arc<T>` and `Box<T>` ([#1277](https://github.com/paritytech/subxt/pull/1277))
```

Comment:
I would say that :point_up: was "feature PR, and it would be more suitable add under "Added"